pcdscalc/misc_calcs.py

```diff
@@ -0,0 +1,351 @@
+import numpy as np
+from .be_lens_calcs import alias
+
+# TODO: find a better number
+_manual_energy = 9.8
+
+
+def set_energy(energy):
+    """
+    Set a global, manual, working energy for doing offline calcs.
+    """
+    global _manual_energy
+    _manual_energy = energy
+
+
+def check_id(material_id):
+    """Check to see if you are using an alias. Return the chemical formula."""
+    try:
+        return alias[material_id]
+    except Exception:
+        return material_id
+
+
+def cosd(angle):
+    """Cos of an angle specified in degrees."""
+    arad = np.deg2rad(angle)
+    x = np.cos(arad)
+    return x
+
+
+def sind(angle):
+    """Sin of an angle specified in degrees."""
+    arad = np.deg2rad(angle)
+    x = np.sin(arad)
+    return x
+
+
+def asind(angle):
+    """Arcsin in degrees. Closed interval of [-pi/2, pi/2]."""
+    a = np.arcsin(angle)
+    a = np.rad2deg(a)
+    return a
+
+
+def get_energy(energy=None, correct_ev=True):
+    """
+    Get working energy.
+
+    Convert to eV if correct_ev == True
+    If energy passed in, return it, otherwise return manual_energy if set, or
+    machine energy otherwise.
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in eV or keV.
+    correct_ev : bool
+        Indicates if energy should be converted to eV.
+
+    Returns
+    -------
+    en : number
+        Photon energy in eV or keV.
+    """
+    en = None
+    if energy is not None:
+        en = energy
+    elif _manual_energy is not None:
+        en = _manual_energy
+    else:
+        # en = _ePv.get()
+        # TODO: we need a way of getting the actual energy here.
+        pass
+    if correct_ev:
+        en = to_ev(en)
+        pass
+    return en
+
+
+def to_ev(energy):
```

Comment:
This is a scary function...

And the `sub_100_energy` thing, oh boy...

Reply:
I think we can stick with one way or the other and have the user be responsible to convert the energy if needed. 
Most of the time we want energy to be in eV anyway, so having functions ask for energy in eV would do it - these two functions can go away

Reply:
Yeah, I'd hope this could be removed. It's something that could cause very strange results unexpectedly, especially if these functions ever start getting called in other library functions (where we have to start passing along `sub_100ev` everywhere...)

Reply:
Cristina sent me the link so let me make some comment related to the history: is this the function that takes anything below 100 as keV unit? Yes that was an XPP decision in the early days, and since we don't care about soft or tender x rays so much, there's not much ambiguity. The problem now is that we want to make hutch python general, and the soft guys want eV as their unit, and the hard guys like keV (less key strokes, that's important!) ... good luck resolving this conflict :)

Reply:
Ohh that's just going to make it more interesting. Thanks a lot for letting use know about this.

Reply:
For library-level methods, I think we have to put our foot down and say "there is only one true energy unit".

If we end up needing to wrap these per-beamline to scale energy inputs in user-facing functions, that'd be scary - but perhaps less so compared to scaling every energy under a seemingly arbitrary threshold?